File: annotationprocessortesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/element/EmptyTag.java
package com.tickaroo.tikxml.annotationprocessing.element;

import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "empty")
public class EmptyTag {

  @Override public boolean equals(Object obj) {
    return obj instanceof EmptyTag;
  }
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/element/EmptyTag.kt
package com.tickaroo.tikxml.annotationprocessing.element

import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "empty")
class EmptyTag {
    override fun equals(other: Any?): Boolean {
        return other is EmptyTag
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}